src/api/app-agent/websocket.ts

```diff
@@ -49,15 +50,31 @@ export class AppAgentWebsocket implements AppAgentClient {
 
   emitter = new Emittery<AppAgentEvents>();
 
-  constructor(appWebsocket: AppWebsocket, installedAppId: InstalledAppId) {
+  private constructor(
+    appWebsocket: AppWebsocket,
+    installedAppId: InstalledAppId
+  ) {
     this.appWebsocket = appWebsocket;
+
     this.installedAppId = installedAppId;
 
     this.appWebsocket.on("signal", (signal) =>
       this.emitter.emit("signal", signal)
     );
   }
 
+  static async connect(
```

Comment:
I don't love calling this connect. Given that the normal usage is 
```
const appWs = AppWebsocket.connect()
const appAgentWs = AppAgentWebsocket.connect(appWs)
```
It's a little weird to say we're connecting twice, really we're connecting once.

What we're really doing is currying, but calling this function `.curry` is probably not great devX :)

Open to suggestions if someone has a better name.